### PR TITLE
1002 cut sym

### PR DESCRIPTION
**Pull request by oerc0122**
Add cut_sym with symops

### CODE REVIEW

File: herbert_core/utilities/maths/merge_ranges.m [hunk 1]
```diff
@@ -0,0 +1,42 @@
+function [starts, lengths] = merge_ranges(starts, lengths)
+% Merges cell arrays of intervals (defined as starts and lengths to read) into
+% single block of starts, lengths eliminating overlap.
```
Comment (cmarooney-stfc) [inline herbert_core/utilities/maths/merge_ranges.m:3]:
```suggestion
% single block of starts, lengths eliminating any overlap between adjacent intervals to create a single compound interval from any set of overlapping integrals.
```

File: horace_core/sqw/@SQWDnDBase/private/cut_parse_inputs_.m [hunk 2]
```diff
@@ -56,19 +56,33 @@
 %                                integration range for three cuts, the first
 %                                cut integrates the axis over 105-107, the
 %                                second over 109-111 and the third 113-115.
+%
+%   sym             Symmetry operator (or an array of symmetry operators
+%                  to be applied in the order sym(1), sym(2),...)
+%                  by which a symmetry related cut is to be accumulated.
```
[the remark this answers is not in the document]
Comment (oerc0122) [reply horace_core/sqw/@SQWDnDBase/private/cut_parse_inputs_.m:62]: The function is called `cut_accumulate_data_` so I presume it's a known thing in the field.

File: horace_core/sqw/@SQWDnDBase/private/cut_parse_inputs_.m [hunk 3]
```diff
@@ -336,20 +375,64 @@
...
+%           A symmetry description can be:
+%           - Scalar symop object
+%           - Array of symop objects (multiple symops to be performed in sequence)
+%           - Empty argument (which will be removed)
+%
+% Output:
+% -------
+%   sym_out Cell array of symmetry descriptions, each one a scalar or row vector
+%           of symop objects. Empty symmetry descriptions or identity descriptions
+%           are removed from the cell array.
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@SQWDnDBase/private/cut_parse_inputs_.m:415]: presuming this is a subset of the input argument, but you don't say so

File: horace_core/sqw/@sqw/private/cut_accumulate_data_.m [hunk 4]
```diff
@@ -55,11 +55,15 @@
     block_starts = 1;
     block_sizes = obj.data.npix;
 else
-    [block_starts, block_sizes] = sproj.get_nrange(obj.data.npix,saxes,targ_axes,targ_proj);
+    block_starts = cell(numel(targ_proj), 1);
+    block_sizes = cell(numel(targ_proj), 1);
+    [block_starts, block_sizes] = arrayfun(@(proj, ax) sproj.get_nrange(obj.data.npix, saxes, ax, proj), ...
```
[the remark this answers is not in the document]
Comment (cmarooney-stfc) [reply horace_core/sqw/@sqw/private/cut_accumulate_data_.m:58]: sorry I had not noticed that these are also input arguments to line 60

File: horace_core/sqw/@sqw/private/cut_accumulate_data_.m [hunk 5]
```diff
@@ -73,7 +77,7 @@
 cut_in_mem = ~obj.pix.is_filebacked || ~PixelDataBase.do_filebacked(n_candidate_pix);
 cut_to_file = ~return_cut;
 
-if ~cut_to_file && ~PixelDataBase.do_filebacked(n_candidate_pix, 10)
+if ~cut_to_file && PixelDataBase.do_filebacked(n_candidate_pix, 10)
```
[the remark this answers is not in the document]
Comment (oerc0122) [reply horace_core/sqw/@sqw/private/cut_accumulate_data_.m:78]: Yes, I did. I want the warning to happen if it would fail over 10*mem_chunk_size

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/horace_function_utils/private/is_horace_data_file_opt.m [hunk 6]
```diff
@@ -1,14 +1,9 @@
...
-        opt_sqw=true;
-    elseif strcmpi(opt,'$dnd')
-        opt_dnd=true;
-    elseif strcmpi(opt,'$hor')
-        opt_hor=true;
-    end
-end
+opt_sqw=strcmpi(opt,'$sqw');
+opt_dnd=strcmpi(opt,'$dnd');
+opt_hor=strcmpi(opt,'$hor');
```
[the remark this answers is not in the document]
Comment (oerc0122) [reply horace_core/horace_function_utils/private/is_horace_data_file_opt.m:5]: I wanted to do this as it doesn't matter if it's `char` or `string` and `strcmpi` is perfectly happy with `strcmpi(13, 'hello')` so it's not a problem.

**Pull request merged**